src/components/CallView/shared/VideoBackground.vue

```diff
@@ -0,0 +1,85 @@
+<!--
+  - @copyright Copyright (c) 2019 Marco Ambrosini <marcoambrosini@pm.me>
+  -
+  - @author Marco Ambrosini <marcoambrosini@pm.me>
+  -
+  - @license GNU AGPL version 3 or any later version
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+-->
+
+<template>
+	<div
+		:style="{'background-color': backgroundColor }"
+		class="video-background">
+		<h3>{{ displayName[0] }}</h3>
+		<p>{{ displayName }}</p>
+	</div>
+</template>
+
+<script>
+export default {
+	name: 'VideoBackground',
+
+	props: {
+		displayName: {
+			type: String,
+			default: '?',
+		},
+	},
+
+	computed: {
+		backgroundColor() {
+			// If the prop is empty. We're not checking for the default value
+			// because the user's displayName might be '?'
+			if (!this.$options.propsData.displayName) {
```

Comment:
I'm not sure I understand this correctly.
Why put a default to '?' and not use it in the end?

Reply:
Also, should be `this.$props`, not `this.$options.propsData` :wink:

Reply:
this.$options.propsData tells me if the prop  has been passed in. So I can differentiate between my default "?" (gray bkg) and a passed in "?" displayname (toRgb color)

Reply:
Anyway in the end I think I will just show the avatar all the time

Reply:
Ah right, I see the difference.
Not sure if it's a reactive property though.